src/Rest/RestApiBrowser.php

```diff
@@ -70,62 +110,48 @@ public function sendRequest($method, $url, $body = null, array $options = array(
         }
 
         if (null !== $this->responseStorage) {
-            $this->responseStorage->writeRawContent($this->response->getBody(true));
+            $this->responseStorage->writeRawContent($this->response->getBody()->getContents());
         }
     }
 
     /**
-     * @param string $name
-     * @param string $value
+     * @param string $method
+     * @param string $uri With or without host
+     * @param string|resource|array $body
      */
-    public function addRequestHeader($name, $value)
+    private function send($method, $uri, $body = null)
     {
-        if (isset($this->requestHeaders[$name])) {
-            if (!is_array($this->requestHeaders[$name])) {
-                $this->requestHeaders[$name] = array($this->requestHeaders[$name]);
-            }
-            $this->requestHeaders[$name][] = $value;
-        } else {
-            $this->requestHeaders[$name] = $value;
+        if (!$this->hasHost($uri)) {
+            $uri = rtrim($this->httpClient->getConfiguration()->getBaseUri(), '/') . '/' . ltrim($uri, '/');
         }
+        $stream = new Stream('php://memory', 'rw');
+        if ($body) {
+            $stream->write($body);
```

Comment:
@tyx, @dsamblas   here body must absolutely be a string in fact... as we use [Zend\Diactoros\Stream::write()](https://github.com/zendframework/zend-diactoros/blob/c3b31a5040bef3f73f0348df607f2a0bd978723a/src/Stream.php#L204) method specified by [StreamInterface:write()](https://github.com/php-fig/http-message/blob/85d63699f0dbedb190bbd4b0d2b9dc707ea4c298/src/StreamInterface.php#L115).

So... if we accept resources & arrays, we have to:
- Convert them into string.
- Add corresponding unit tests for that.